table/block_based_table_reader.cc

```diff
@@ -1639,9 +1648,11 @@ BlockIter* BlockBasedTable::NewDataBlockIterator(
 
   if (s.ok()) {
     assert(block.value != nullptr);
-    iter = block.value->NewIterator(
-        &rep->internal_comparator, rep->internal_comparator.user_comparator(),
-        iter, true, rep->ioptions.statistics, key_includes_seq);
+    const bool kTotalOrderSeek = true;
+    iter = block.value->NewIndexOrDataIterator(
+        is_index, &rep->internal_comparator,
+        rep->internal_comparator.user_comparator(), iter,
+        rep->ioptions.statistics, kTotalOrderSeek, key_includes_seq);
```

Comment:
Same question here. Do we ever get index block in this function at all? If not then we don't need this NewIndexOrDataIterator() function.

Reply:
partitioned index blocks